### ISSUE TITLE
How to run an untargeted attack?

### ISSUE THREAD

**Issue by GaoJiqiang** [rtaori/Black-Box-Audio#13]
I want to perform untargeted attack, how should i adjust the code?

**Comment by rtaori** [comment]
Hi,
You should pass in the original correct transcription as the target phrase, and then change [this line](https://github.com/rtaori/Black-Box-Audio/blob/master/run_audio_attack.py#L203) to select the lowest scoring samples from the population (instead of the highest scoring ones).

**Comment by GaoJiqiang** [comment]
> Hi,
> You should pass in the original correct transcription as the target phrase, and then change [this line](https://github.com/rtaori/Black-Box-Audio/blob/master/run_audio_attack.py#L203) to select the lowest scoring samples from the population (instead of the highest scoring ones).

Thanks for your reply, it works. Your work is impressive! One more question I have is that if I want to test other cloud ASR systems (e.g., Google, IBM), the CTC_score is unobtainable. Should I find or build another fitness_score function? Do you think it will work? They often return only transcripts, some with confidence score.

**Comment by rtaori** [comment]
Great to know it works! Ah, I see. In our case, our attack assumes you have access to the output logits of the model. If you only have hard transcripts, you'd need to adapt the attack in that case (and it probably won't work as well). There may be some existing research attempting to tackle the problem you're suggesting, but I'm not sure. Sorry I can't be more helpful.
